app/controllers/api/v1/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Api
+  module V1
+    class ProductsController < ApplicationController
+      before_action :set_product, only: %i[show update destroy]
+
+      def index
+        render json: { data: Product.all }
+      end
+
+      def show
+        render json: @product
+      end
+
+      def create
+        @product = Product.new(product_params)
+        if @product.save
+          render json: @product
+        else
+          render json: { errors: @product.errors.full_messages.join(', ') }, status: :unprocessable_entity
+        end
+      end
+
+      def update
+        if @product.update(product_params)
+          render json: @product
+        else
+          render json: { errors: @product.errors.full_messages.join(', ') }, status: :unprocessable_entity
+        end
+      end
+
+      def destroy
+        @product.destroy!
+        render json: @product
+      rescue RecordNotSaved
+        render json: { Message: 'Something went wrong!' }, status: :unprocessable_entity
```

Comment:
JSON keys should be lowercase
```suggestion
        render json: { message: 'Something went wrong!' }, status: :unprocessable_entity
```